app/main/services/process_request_json.py

```diff
@@ -51,8 +51,28 @@ def _hash_to(arguments, document):
         document[target_field] = hashlib.sha256((six.text_type(document[source_field])).encode('utf-8')).hexdigest()
 
 
+def _update_conditionally(arguments, document):
```

Comment:
I don't think I have a problem with this thing existing, but seeing as the implementation is almost the same as `append_conditionally`, we should probably re-implement that function as a call to this one (but with some parameter set e.g. `append=True`).

Obviously there are ways of implementing this that avoid inserting an `if`, but more complicated might not be very beneficial here.

Reply:
@galund Have refactored!